### PR TITLE
xds: better way to compare and update dropper in EDS

### PR DESCRIPTION
**Pull request by menghanl**

### CODE REVIEW

File: xds/internal/balancer/edsbalancer/eds_impl.go [hunk 1]
```diff
@@ -145,42 +146,24 @@ func (edsImpl *edsBalancerImpl) HandleChildPolicy(name string, config json.RawMe
 // updateDrops compares new drop policies with the old. If they are different,
 // it updates the drop policies and send ClientConn an updated picker.
 func (edsImpl *edsBalancerImpl) updateDrops(dropPolicies []xdsclient.OverloadDropConfig) {
```
Comment (easwars) [inline xds/internal/balancer/edsbalancer/eds_impl.go]: Nit: Should the input parameter be renamed to `dropConfig` instead?

**Comment by easwars** [comment]
Not directly related to the changes made in this PR:
I see comments in `internal/wrr/wrr.go` that `Next` needs to be thread safe. It doesn't say that `Add` needs to be thread-safe. Eitherways, the randomWRR implementation does not seem to be thread-safe. I think our usage here is fine since we only ever call `Add` during the creation of that WRR picker, and after that we only call `Next`. But it does look like the wrr package needs fixing. Wanted to confirm with you before filing an issue.

File: xds/internal/balancer/edsbalancer/eds_impl.go (hunk 1 above)
Comment (menghanl) [reply xds/internal/balancer/edsbalancer/eds_impl.go]: Done

**Comment by menghanl** [comment]
On `wrr`, sounds right. We should fix randomWRR and also the comment.

`Add` was expected to be called only to build `wrr` (so `Add` is like balancer's `handleUpdate` function, `Next` is like `Pick`). But since `Add` is a method in the same interface as `Next`, there's no way stopping the caller from calling `Add` concurrently with `Next`.

Another option is to remove `Add` from the interface, and make the constructor take a list of items. But this sounds quite limited.

**Pull request merged**